rohmu/object_storage/s3.py

```diff
@@ -542,12 +543,11 @@ def check_or_create_bucket(self) -> None:
         try:
             self.s3_client.head_bucket(Bucket=self.bucket_name)
         except botocore.exceptions.ClientError as ex:
+            # https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html
             status_code = ex.response.get("ResponseMetadata", {}).get("HTTPStatusCode")
             if status_code == 301:
                 raise InvalidConfigurationError(f"Wrong region for bucket {self.bucket_name}, check configuration")
-            elif status_code == 403:
-                self.log.warning("Access denied on bucket check, assuming write permissions")
-            elif status_code == 404:
+            elif status_code in {HTTPStatus.BAD_REQUEST, HTTPStatus.FORBIDDEN, HTTPStatus.NOT_FOUND}:
```

Comment:
Just to keep track of the discussion: in some places we rely on the current behaviour for 403/Forbidden of not creating a bucket and simply logging a message, so need to double check how to fix this part without breaking other stuff.